co-noir/ultrahonk/src/decider/polynomial.rs

```diff
@@ -0,0 +1,210 @@
+use ark_ff::PrimeField;
+use num_traits::Zero;
+use std::ops::{AddAssign, Index, IndexMut};
+
+#[derive(Clone, Debug, Default)]
+pub struct Polynomial<F> {
+    pub coefficients: Vec<F>,
+}
+
+pub struct ShiftedPoly<'a, F> {
+    pub(crate) coefficients: &'a [F],
+    zero: F, // TODO is there are better solution
```

Comment:
that is annoying and only really needed for index below, right?

Reply:
let me think about some solutions

Reply:
Discussed this with Fabian when I implemented this, we did not find a clean solution quickly, so we did this workaround. Maybe something with LazyCell? Biggest problem is the generic F.

Reply:
yeah we could maybe do some hacky workarounds but lets just leave it for the moment